krun/vm_defs.py

```diff
@@ -15,26 +19,63 @@
 BASE_ENV = os.environ.copy()
 BASE_ENV.update({"LD_LIBRARY_PATH": os.path.join(DIR, "..", "libkruntime")})
 
-class BaseVMDef(object):
 
-    def __init__(self, iterations_runner, extra_env=None):
+class EnvChange(object):
+    def __init__(self, var, val):
+        self.var, self.val = var, val
+
+class EnvChangeSet(EnvChange):
+    pass
+
+class EnvChangeAppend(EnvChange):
+    pass
+
+
+def _apply_env_changes(env, changes):
```

Comment:
it looks to me like it would be much more natural to make these changes be methods on the EnvChange classes, instead of the isinstance checks here.

Also, I think there needs to be a docstring in some of these classes to explain what they are for.